slack-base/src/main/java/com/hubspot/slack/client/models/events/bot/SlackAppHomeOpenedEventIF.java

```diff
@@ -0,0 +1,36 @@
+package com.hubspot.slack.client.models.events.bot;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.hubspot.immutables.style.HubSpotStyle;
+import com.hubspot.slack.client.methods.interceptor.HasUser;
+import com.hubspot.slack.client.models.events.SlackEvent;
+import com.hubspot.slack.client.models.response.views.HomeTabViewResponse;
+import org.immutables.value.Value;
+
+import java.util.Optional;
+
+@Value.Immutable
+@HubSpotStyle
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@JsonDeserialize(as = SlackAppHomeOpenedEvent.class)
+public interface SlackAppHomeOpenedEventIF extends SlackEvent, HasUser {
+    String getTab();
+
+    Optional<HomeTabViewResponse> getView();
+
+    @JsonProperty("user")
+    String getUserId();
+
+    @JsonProperty("channel")
+    String getChannelId();
+
+    //Home opened events do not have a ts, so we manually set it as null
```

Comment:
This makes me nervous, since it violates our implicit invariant that any nullable field is wrapped in`Optional` and any non-`Optional` field is not `null`. What it suggests to me is that our initial class hierarchy was not correct, and that app events (i.e. things that may lack a `ts` field`) belong in a slightly different inheritance tree.

I haven't dug deeply into the difference in shapes, but I could imagine having a base `SlackEvent` that contains all common fields, and then another `HasReferencedMessage` interface that adds the `ts` field. We could add an intermediate `SlackMessageEvent`to capture slack events which also have a `ts` field, and then we wouldn't need to do this hack.

Does that make sense?

Reply:
Makes total sense to me. I was just following the pattern that was already there for some other events. I'll see if I can shuffle things around like you suggest, but it might be a few days and it also might be little out of my comfort zone, though I'm starting to get more and more familiar with this codebase :)

Reply:
If this is how things already are and this is blocking you, we can go ahead and copy the existing pattern to get you moving. If you don't mind taking a stab at this style of cleanup, if you find it's not too onerous we'd appreciate it! :)

Thanks!

Reply:
See my comment in issue #128 re the class cleanup. I'm already on my way (I have a fork I'm using in prod), so your call on whether you want to commit this or wait for the bigger cleanup on `ts`. I just don't know how much I can help on that front.